File: src/main/java/com/jradams/melophobia/entity/Issue.java
package com.jradams.melophobia.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinTable;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.ManyToOne;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import java.util.Set;

@Entity
@Getter
@Setter
@NoArgsConstructor
@ToString
public class Issue {

    @Id
    @Column(name = "issue_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long issueId;

    @ManyToOne
    @JoinColumn(name = "label_id")
    private Label label;

    @ManyToOne
    @JoinColumn(name = "release_id")
    private Release release;

    @Pattern(regexp = "\\d{4}-\\d{2}-\\d{2}", message = "Release date format: '####-##-##'")
    private String releaseDate;

    @NotBlank(message = "Catalogue ID is mandatory")
    private String catalogueId;

    @ManyToMany
    @JoinTable(name = "issue_countries", joinColumns = @JoinColumn(name = "issue_id"),
            inverseJoinColumns = @JoinColumn(name = "country_id"))
    private Set<Country> countries;

    @NotBlank(message = "Edition is mandatory")
    private String edition;

    private Boolean isRerelease;
    private Boolean isOfficial;
    private String barcode;

    @ManyToOne
    @JoinColumn(name = "media_id")
    private Media media;

    private String matrixRunout;
    private String description;
}

File: Service/Model/Issue.cs
namespace Melophobia.Model
{
    public partial class Issue
    {
        public int Id { get; init; }

        public int ReleaseId { get; init; }

        public int LabelId { get; init; }

        public string? ReleaseDate { get; init; }

        public string? CatalogueNumber { get; init; }

        public int? MediaId { get; init; }

        public int? Edition { get; init; }

        public bool? IsReissue { get; init; }

        public bool? IsOfficial { get; init; }

        public bool? IsPromotional { get; init; }

        public bool? HasPregapTrack { get; init; }

        public bool? HasDataTrack { get; init; }

        public string? Barcode { get; init; }

        public string? Comment { get; init; }

        public virtual ICollection<CollectionItem> CollectionItems { get; init; } = new List<CollectionItem>();

        public virtual ICollection<IssueVariant> IssueVariants { get; init; } = new List<IssueVariant>();

        public virtual Label Label { get; init; } = null!;

        public virtual Medium? Media { get; init; }

        public virtual Release Release { get; init; } = null!;

        public virtual ICollection<Country> Countries { get; init; } = new List<Country>();

        public virtual ICollection<Track> Tracks { get; init; } = new List<Track>();
    }
}
